src/sentry/spans/consumers/process/factory.py

```diff
@@ -86,6 +90,16 @@ def _build_snuba_span(relay_span: Mapping[str, Any]) -> MutableMapping[str, Any]
 
     snuba_span["sentry_tags"] = sentry_tags
 
+    event_data = {}
+    event_data["transaction"] = sentry_tags.get("transaction", "")
+    event_data["contexts"] = {"trace": {"span_id": snuba_span["span_id"]}}
+    event_data["spans"] = [deepcopy(snuba_span)]
```

Comment:
Not 100% sure because I don't know the grouping code, but when `is_segment` is `true`, should we leave `event_data["spans"]` empty to simulate a transaction?

Reply:
Good point.

Reply:
Oh, why does segment ingestion affect span grouping? Does this mean we'll _never_ have `group_raw`? If we'll never have it, we can't link to the right span in a trace, which isn't great UX

Reply:
I would still call `execute_strategy` for segments (i.e. transactions). I just would not put anything in `event_data["spans"]` in that case. Because a pseudo-event that contains itself as a span seems wrong.

Reply:
But then we won't have a `span_id` to get the group, which means it'll be `"0"`.

Reply:
That sounds right! If a segment is roughly the same as a transaction, the strategy will give it a group ID based on the transaction name: https://github.com/getsentry/sentry/blob/f4144886a747d651ce7472cdf3c9beaf0ac62c23/src/sentry/spans/grouping/strategy/base.py#L47-L50

Reply:
@jjbayer are we getting the transaction name when we're receiving a segment? as description or a Sentry tag maybe?